newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -137,221 +54,222 @@ type State = {|
+  });
+
+  if (
+    !optionList.filter(option => option.type !== 'separator' && option.value)
+      .length
+  )
+    return [];
+
+  // Remove divider(s) if they are at the start or end of array
+  while (
+    optionList[optionList.length - 1] !== undefined &&
+    optionList[optionList.length - 1].type !== undefined
+  )
+    optionList.pop();
+  while (optionList[0] !== undefined && optionList[0].type !== undefined)
+    optionList.shift();
+
+  return optionList;
+};
+
+const handleChange = (
+  event: SyntheticKeyboardEvent<HTMLInputElement>,
+  option: Option | string,
+  props: Props,
+  ref: { current: ?HTMLInputElement }
+): void => {
+  if (option !== null && option.type !== 'separator') {
+    if (typeof option === 'string') props.onChange(option);
+    else {
+      if (option.onClick) option.onClick();
+      else
+        props.onChoose
+          ? props.onChoose(option.value)
+          : props.onChange(option.value);
+    }
   }
+  if (event.key === 'Enter' && ref.current) ref.current.blur();
+};
+
+const getDefaultStylingProps = (
+  params: Object,
+  value: string,
+  props: Props,
+  ref: { current: ?HTMLInputElement }
+): Object => {
+  const { InputProps, inputProps, InputLabelProps, ...other } = params;
+  return {
+    ...other,
+    InputProps: {
+      ...InputProps,
+      className: null,
+      endAdornment: null,
+    },
+    inputProps: {
+      ...inputProps,
+      className: null,
+      onKeyDown: (event: SyntheticKeyboardEvent<HTMLInputElement>): void => {
+        if (event.key === 'Enter') handleChange(event, value, props, ref);
+      },
+    },
+  };
+};
+
+const getOptionLabel = (option: Option, value: string): string =>
+  option.value ? option.value : value;
+
+export default React.forwardRef<Props, SemiControlledAutoCompleteInterface>(
+  function SemiControlledAutoComplete(props: Props, ref) {
+    const input = React.useRef((null: ?HTMLInputElement));
+    const [state, setState] = useState<State>({ inputValue: null });
+    const classes = useStyles();
+
+    React.useImperativeHandle(ref, () => ({
+      focus: () => {
+        if (input.current) input.current.focus();
+      },
+      forceInputValueTo: (newValue: string) => {
+        if (state.inputValue !== null) setState({ inputValue: newValue });
+      },
+    }));
 
-  render() {
-    const { props, state } = this;
     const currentInputValue =
       state.inputValue !== null ? state.inputValue : props.value;
 
     const helperText = props.helperMarkdownText ? (
       <MarkdownText source={props.helperMarkdownText} />
     ) : null;
 
+    const handleInputChange = (
+      event: SyntheticKeyboardEvent<HTMLInputElement>,
+      value: string,
+      reason: string
+    ): void => setState({ inputValue: value });
+
     return (
       <I18n>
         {({ i18n }) => (
-          <Downshift
+          <Autocomplete
+            freeSolo
+            blurOnSelect
```

Comment:
Any reason for blurOnSelect? I think we should remove it - it's important to keep the focus after a selection I think, in particular when selecting an object in a list of parameters, where you want to use Tab to go to the next field

(For example:

![image](https://user-images.githubusercontent.com/1280130/76792347-26797880-67c3-11ea-8f73-60cefe953108.png)

after selecting the object, we should be able to Tab to go to "Yes" button)

Reply:
I had added keybinding for 'Enter' which would let users select anything they type. If it was something arbitrary, the menu wouldn't close, so I had to blur it. To make it consistent, I added blurOnSelect. Now I guess I'd remove the keybinding to Enter (it's unnecessary at the moment) and use blur event to select arbitrary values like it used to happen with downshift :)

Reply:
> Now I guess I'd remove the keybinding to Enter (it's unnecessary at the moment) and use blur event to select arbitrary values like it used to happen with downshift

Sounds good. Usage of the blur event is only because we're a "SemiControlledAutoComplete" - the "SemiControlled" means that it's keeping its state internally until it's blurred :) 

We definitely want to remove blurOnSelect - losing the focus would be a usability bug.
Removing the keybinding to Enter also sounds good :) Because I think it should be handled by the material-ui autocomplete. 

The "freeSolo" example on https://material-ui.com/components/autocomplete/ is how the autocomplete should behave I think